src/Microsoft.ML.Core/Data/Schema.cs

```diff
@@ -74,21 +76,35 @@ public Column GetColumnOrNull(string name)
             return null;
         }
 
+        public IEnumerator<Column> GetEnumerator() => ((IEnumerable<Column>)_columns).GetEnumerator();
+
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
         public override string ToString()
         {
-            return $"{ColumnCount} columns";
+            return $"{Count} columns";
         }
 
         /// <summary>
-        /// This class describes one column in the schema.
+        /// This class describes one column in the particular schema.
         /// </summary>
-        public sealed class Column
+        public class Column
```

Comment:
>public class [](start = 8, length = 12)

I strongly object to making this class extensible. Some classes might benefit from polymorphism but a description of a column is *so* fundamental to anyone understanding what the dataview/transformer/whatever actually is that we ought to be able to make very definite statements.

(Also, once we fix that, why not struct?)

Reply:
Oh, right, that's how it happened. I made it a `struct`, then reverted, but didn't put `sealed` back in. I'll try to convert to `struct` now

---
In reply to: [235109399](https://github.com/dotnet/machinelearning/pull/1680#discussion_r235109399) [](ancestors = 235109399)